Crash when exporting `globalThis` in `declare global` block

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  3.4.0-dev.201xxxxx

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**

**Code**

```ts
declare global {
  export { globalThis as global };
}
```

**Expected behavior:**

pass

**Actual behavior:**

```
$ node built/local/tsc --noEmit index.ts
...\TypeScript\built\local\tsc.js:94648
                throw e;
                ^

TypeError: Cannot read property '0' of undefined
```
**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

@falsandtru can you explain why this should pass? The errors I get after fixing the crash look correct:

```
welove.ts:1:9 - error TS2669: Augmentations for the global scope can only be directly nested in external modules or ambient module declarations.

1 declare global {
          ~~~~~~

welove.ts:2:14 - error TS2661: Cannot export 'globalThis'. Only local declarations can be exported from a module.

2     export { globalThis as global }
               ~~~~~~~~~~


Found 2 errors.
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -32454,7 +32454,7 @@ namespace ts {
                 // find immediate value referenced by exported name (SymbolFlags.Alias is set so we don't chase down aliases)
                 const symbol = resolveName(exportedName, exportedName.escapedText, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace | SymbolFlags.Alias,
                     /*nameNotFoundMessage*/ undefined, /*nameArg*/ undefined, /*isUse*/ true);
-                if (symbol && (symbol === undefinedSymbol || isGlobalSourceFile(getDeclarationContainer(symbol.declarations[0])))) {
+                if (symbol && (symbol === undefinedSymbol || symbol === globalThisSymbol || isGlobalSourceFile(getDeclarationContainer(symbol.declarations[0])))) {
                     error(exportedName, Diagnostics.Cannot_export_0_Only_local_declarations_can_be_exported_from_a_module, idText(exportedName));
                 }
                 else {

--- a/src/services/findAllReferences.ts
+++ b/src/services/findAllReferences.ts
@@ -1395,8 +1395,10 @@ namespace ts.FindAllReferences.Core {
                 || exportSpecifier.name.originalKeywordKind === SyntaxKind.DefaultKeyword;
             const exportKind = isDefaultExport ? ExportKind.Default : ExportKind.Named;
             const exportSymbol = Debug.assertDefined(exportSpecifier.symbol);
-            const exportInfo = Debug.assertDefined(getExportInfo(exportSymbol, exportKind, state.checker));
-            searchForImportsOfExport(referenceLocation, exportSymbol, exportInfo, state);
+            const exportInfo = getExportInfo(exportSymbol, exportKind, state.checker);
+            if (exportInfo) {
+                searchForImportsOfExport(referenceLocation, exportSymbol, exportInfo, state);
+            }
         }
 
         // At `export { x } from "foo"`, also search for the imported symbol `"foo".x`.

--- a/tests/baselines/reference/globalThisGlobalExportAsGlobal.errors.txt
+++ b/tests/baselines/reference/globalThisGlobalExportAsGlobal.errors.txt
@@ -1,0 +1,14 @@
+tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts(2,9): error TS2669: Augmentations for the global scope can only be directly nested in external modules or ambient module declarations.
+tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts(3,14): error TS2661: Cannot export 'globalThis'. Only local declarations can be exported from a module.
+
+
+==== tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts (2 errors) ====
+    // https://github.com/microsoft/TypeScript/issues/33754
+    declare global {
+            ~~~~~~
+!!! error TS2669: Augmentations for the global scope can only be directly nested in external modules or ambient module declarations.
+        export { globalThis as global }
+                 ~~~~~~~~~~
+!!! error TS2661: Cannot export 'globalThis'. Only local declarations can be exported from a module.
+    }
+    

--- a/tests/baselines/reference/globalThisGlobalExportAsGlobal.js
+++ b/tests/baselines/reference/globalThisGlobalExportAsGlobal.js
@@ -1,0 +1,8 @@
+//// [globalThisGlobalExportAsGlobal.ts]
+// https://github.com/microsoft/TypeScript/issues/33754
+declare global {
+    export { globalThis as global }
+}
+
+
+//// [globalThisGlobalExportAsGlobal.js]

--- a/tests/baselines/reference/globalThisGlobalExportAsGlobal.symbols
+++ b/tests/baselines/reference/globalThisGlobalExportAsGlobal.symbols
@@ -1,0 +1,10 @@
+=== tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts ===
+// https://github.com/microsoft/TypeScript/issues/33754
+declare global {
+>global : Symbol(global, Decl(globalThisGlobalExportAsGlobal.ts, 0, 0))
+
+    export { globalThis as global }
+>globalThis : Symbol(globalThis)
+>global : Symbol(global, Decl(globalThisGlobalExportAsGlobal.ts, 2, 12))
+}
+

--- a/tests/baselines/reference/globalThisGlobalExportAsGlobal.types
+++ b/tests/baselines/reference/globalThisGlobalExportAsGlobal.types
@@ -1,0 +1,10 @@
+=== tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts ===
+// https://github.com/microsoft/TypeScript/issues/33754
+declare global {
+>global : typeof global
+
+    export { globalThis as global }
+>globalThis : typeof globalThis
+>global : typeof globalThis
+}
+

--- a/tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts
+++ b/tests/cases/conformance/es2019/globalThisGlobalExportAsGlobal.ts
@@ -1,0 +1,4 @@
+// https://github.com/microsoft/TypeScript/issues/33754
+declare global {
+    export { globalThis as global }
+}

--- a/tests/cases/fourslash/documentHighlightsInvalidGlobalThis.ts
+++ b/tests/cases/fourslash/documentHighlightsInvalidGlobalThis.ts
@@ -1,0 +1,9 @@
+/// <reference path="fourslash.ts" />
+
+////declare global {
+////    export { globalThis as [|global|] }
+////}
+
+for (const r of test.ranges()) {
+    verify.documentHighlightsOf(r, [r]);
+}

